pandas/io/sql.py

```diff
@@ -1833,6 +1837,15 @@ def drop_table(self, table_name: str, schema: str | None = None) -> None:
             self.get_table(table_name, schema).drop(bind=self.con)
             self.meta.clear()
 
+    def trunc_table(self, table_name: str, schema: str | None = None) -> None:
+        schema = schema or self.meta.schema
+        if self.has_table(table_name, schema):
+            self.meta.reflect(bind=self.con, only=[table_name], schema=schema)
+            if schema:
+                schema = schema + "."
+            self.execute(f"DELETE FROM {schema or ''}{table_name}")
```

Comment:
Shouldn't this just be TRUNCATE?

Reply:
I used DELETE since sqlite doesn't support TRUNCATE. However, since I had the if statement already for the schema, it will now use DELETE for the schema-less sqlite and use the faster TRUNCATE for everything else.